src/_P078_Eastron.ino

```diff
@@ -173,229 +187,812 @@ boolean Plugin_078(uint8_t function, struct EventStruct *event, String& string)
+          case 1: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM220_SDM120CT_SDM120; break;
+          case 2: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM230; break;
+          case 3: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM630; break;
+          case 4: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM72D; break;
+          case 5: nrOptions = P078_NR_OUTPUT_OPTIONS_DDM18SD; break;
+          case 6: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM72D_V2; break;
         }
+        const __FlashStringHelper *options[nrOptions];
 
+        for (int i = 0; i < nrOptions; ++i) {
+          options[i] = p078_getQueryString(i, P078_MODEL);
+        }
 
-        success = true;
-        break;
+        for (uint8_t i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+          const uint8_t pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+          sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, nrOptions, options);
+        }
       }
 
-    case PLUGIN_WEBFORM_SAVE:
-      {
-          // Save output selector parameters.
-          for (uint8_t i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
-            const uint8_t pconfigIndex = i + P078_QUERY1_CONFIG_POS;
-            const uint8_t choice = PCONFIG(pconfigIndex);
-            sensorTypeHelper_saveOutputSelector(event, pconfigIndex, i, p078_getQueryValueString(choice));
-          }
+      success = true;
+      break;
+    }
 
-          P078_DEV_ID = getFormItemInt(P078_DEV_ID_LABEL);
-          P078_MODEL = getFormItemInt(P078_MODEL_LABEL);
-          P078_BAUDRATE = getFormItemInt(P078_BAUDRATE_LABEL);
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      // Save output selector parameters.
+      int model = P078_MODEL;
 
-          Plugin_078_init = false; // Force device setup next time
-          success = true;
-          break;
+      for (uint8_t i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
+        const uint8_t pconfigIndex = i + P078_QUERY1_CONFIG_POS;
+        const uint8_t choice       = PCONFIG(pconfigIndex);
+        sensorTypeHelper_saveOutputSelector(event, pconfigIndex, i, p078_getQueryValueString(choice, model));
       }
 
+      P078_DEV_ID   = getFormItemInt(P078_DEV_ID_LABEL);
+      P078_MODEL    = getFormItemInt(P078_MODEL_LABEL);
+      P078_BAUDRATE = getFormItemInt(P078_BAUDRATE_LABEL);
+
+      Plugin_078_init = false; // Force device setup next time
+      success         = true;
+      break;
+    }
+
     case PLUGIN_INIT:
-      {
-        if (Plugin_078_SoftSerial != nullptr) {
-          delete Plugin_078_SoftSerial;
-          Plugin_078_SoftSerial=nullptr;
-        }
-        Plugin_078_SoftSerial = new (std::nothrow) ESPeasySerial(static_cast<ESPEasySerialPort>(CONFIG_PORT), CONFIG_PIN1, CONFIG_PIN2);
-        if (Plugin_078_SoftSerial == nullptr) {
-          break;
-        }
-        unsigned int baudrate = p078_storageValueToBaudrate(P078_BAUDRATE);
-        Plugin_078_SoftSerial->begin(baudrate);
+    {
+      Plugin_078_init = true;
```

Comment:
init should be set to true when all steps are complete.
Now you're performing all actions which may fail after setting the init to true.

Reply:
It would be nice if you can help me out a bit.
Thanks a lot in advance